src/normalizers/keywordProtector.js

```diff
@@ -0,0 +1,49 @@
+import { regexMatchingDecoderMaker } from './regex.js';
+
+export default class KeywordProtector {
+    constructor(keywordList, keywordMap) {
+        this.keywordList = keywordList;
+        this.keywordMap = keywordMap;
+        this.defaultEncoder = this.regexMatchingEncoderMaker(
+            'protectedStringsDecoder',
+            /(?<protector>protector:\w+)/g,
+            this.keywordList.reduce((out, str) => { out[`protector:${str}`] = str; return out}, {})
```

Comment:
you could just do `Object.fromEntries(Object.keys(keywordMap).map(kw => [ `protector:${kw}`, kw ])`

Reply:
Cool!  Changed.